### PR TITLE
[PL examples]: fix progress bar bug

### PR DESCRIPTION
**Pull request by hugoabonizio**
As shown by @prabalbansal in #3576, the fine tuning script for seq2seq models is failing with the following error:

This PR includes the fix suggested by @sshleifer in [this comment](https://github.com/huggingface/transformers/issues/3576#issuecomment-611755174), which made it work, but it seems to be suboptimal since the loss is not shown in the progress.

### CODE REVIEW

File: examples/transformer_base.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def optimizer_step(self, epoch, batch_idx, optimizer, optimizer_idx, second_orde
         self.lr_scheduler.step()
 
     def get_tqdm_dict(self):
-        tqdm_dict = {"loss": "{:.3f}".format(self.trainer.avg_loss), "lr": self.lr_scheduler.get_last_lr()[-1]}
+        avg_loss = getattr(self.trainer, "avg_loss", 0.0)
```
Comment (sshleifer) [inline examples/transformer_base.py:107]: @williamFalcon 
Is there a cleaner way to fix this?

Comment (williamFalcon) [reply examples/transformer_base.py:107]: i think this is outdated. We should deprecate  this function.

Instead use:

```python

bar = {...}
return {"loss": loss, "progress_bar": bar}
```
